### PR TITLE
Allow discovery to be removed, and add peers explicitly to peer store

### LINKED ISSUE #2400: --no-discovery flag not registered
```
[2019-04-27 10:15:36] ERROR main: flag provided but not defined: -no-discovery
```

The above error occurs despite the flag being listed in the `--help` flag options.

### PR DESCRIPTION
**Pull request by atoulme**
Resolves #2400
Also adds the ability to inject static peers from command line.

### CODE REVIEW

File: shared/p2p/service.go [hunk 1]
```diff
@@ -38,6 +38,8 @@ const prysmProtocolPrefix = "/prysm/0.0.0"
 // full beacon states over the wire for our current implementation.
 const maxMessageSize = 1 << 24
 
+const oneHour = time.Duration(1) * time.Hour
```
Comment (prestonvanloon) [inline shared/p2p/service.go]: This is only used once, please remove this const

File: shared/cmd/flags.go [hunk 2]
```diff
@@ -53,6 +53,11 @@ var (
 		Name:  "no-discovery",
 		Usage: "Enable only local network p2p and do not connect to cloud bootstrap nodes.",
 	}
+	// StaticPeers specifies a set of peers to connect to explicitly.
+	StaticPeers = cli.StringSliceFlag{
+		Name:  "peer",
```
Comment (prestonvanloon) [inline shared/cmd/flags.go]: Does this accept a comma separated list of peers? 

Please update the name and/or description. Reading this makes me think I would write `--peer=hostA:1234,hostB:1234,hostC:1234` but i'm not sure if it is supposed to `--peer=hostA:1234 --peer=hostB:1234 --peer=hostC:1234`

File: shared/p2p/service.go [hunk 3]
```diff
@@ -173,29 +178,37 @@ func (s *Server) Start() {
...
 	}
 
-	if !s.noDiscovery {
-		startPeerWatcher(ctx, s.host, s.bootstrapNode, s.relayNodeAddr)
+	for _, peer := range s.staticPeers {
+		peerInfo, err := peerInfoFromAddr(peer)
+		if err != nil {
+			log.Errorf("Invalid peer address: %v", err)
+		} else {
+			s.host.Peerstore().AddAddrs(peerInfo.ID, peerInfo.Addrs, oneHour)
```
Comment (prestonvanloon) [inline shared/p2p/service.go]: Why one hour? Shouldn't it be indefinitely?

File: shared/p2p/service.go (hunk 1 above)
Comment (atoulme) [reply shared/p2p/service.go]: ok

File: shared/cmd/flags.go (hunk 2 above)
Comment (atoulme) [reply shared/cmd/flags.go]: I tried it manually. Indeed right now it's the latter: `--peer=hostA:1234` `--peer=hostB:1234` `--peer=hostC:1234`. Would you rather like to use a comma separated entry? Please advise. We may be able to combine both possibilities.

File: shared/p2p/service.go (hunk 3 above)
Comment (atoulme) [reply shared/p2p/service.go]: I profess my ignorance of the API here. I assumed a TTL was mandatory.

Comment (atoulme) [reply shared/p2p/service.go]: It does seem like a TTL is mandatory.

File: shared/cmd/flags.go (hunk 2 above)
Comment (prestonvanloon) [reply shared/cmd/flags.go]: OK great I understand. This is OK

File: shared/cmd/flags.go [hunk 4]
```diff
@@ -53,6 +53,11 @@ var (
...
+		Usage: "Connect with these peers.",
```
Comment (prestonvanloon) [inline shared/cmd/flags.go]:
```suggestion
		Usage: "Connect with this peer. This flag may be used multiple times.",
```

File: shared/p2p/service.go (hunk 1 above)
Comment (prestonvanloon) [reply shared/p2p/service.go]:
```suggestion
```

File: shared/cmd/flags.go (hunk 2 above)
Comment (atoulme) [reply shared/cmd/flags.go]: I have a patch for you. Just a sec.

File: shared/p2p/service.go (hunk 3 above)
Comment (prestonvanloon) [reply shared/p2p/service.go]:
```suggestion
			s.host.Peerstore().AddAddrs(peerInfo.ID, peerInfo.Addrs, 24 * time.Hour)
```
Yeah, unfortunate design of the API that you can't set a non-expiring TTL. If you provide `0` then it won't add the address at all. Let's set 24hr and assume that is enough

File: shared/p2p/service.go [hunk 5]
```diff
@@ -185,29 +188,39 @@ func (s *Server) Start() {
...
 
-	if err := startmDNSDiscovery(ctx, s.host); err != nil {
-		log.Errorf("Could not start peer discovery via mDNS: %v", err)
-		return
+		startPeerWatcher(ctx, s.host, s.bootstrapNode, s.relayNodeAddr)
 	}
 
-	if !s.noDiscovery {
-		startPeerWatcher(ctx, s.host, s.bootstrapNode, s.relayNodeAddr)
+	maxTime := time.Duration(1 << 62)
```
Comment (prestonvanloon) [inline shared/p2p/service.go]: 👍

File: shared/cmd/flags.go (hunk 4 above)
Comment (atoulme) [reply shared/cmd/flags.go]: ack

**Pull request merged**